extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -741,6 +838,127 @@ public static Table readPartitionedTableWithMetadata(
+            throw new IllegalArgumentException("No partitioning columns");
+        }
+        return readPartitionedTable(new ParquetKeyValuePartitionedLayout(directory, tableDefinition), readInstructions,
+                tableDefinition);
+    }
+
+    /**
+     * Creates a partitioned table via the flat parquet files from the root {@code directory}, inferring the table
+     * definition from those files.
+     *
+     * <p>
+     * Callers wishing to be more explicit and skip the inference step may prefer to call
+     * {@link #readFlatPartitionedTable(File, ParquetInstructions, TableDefinition)}.
+     *
+     * @param directory the source of {@link ParquetTableLocationKey location keys} to include
+     * @param readInstructions the instructions for customizations while reading
+     * @return the table
+     * @see #readPartitionedTable(TableLocationKeyFinder, ParquetInstructions)
+     * @see ParquetFlatPartitionedLayout#ParquetFlatPartitionedLayout(File)
+     */
+    public static Table readFlatPartitionedTable(
+            @NotNull final File directory,
+            @NotNull final ParquetInstructions readInstructions) {
+        return readPartitionedTable(new ParquetFlatPartitionedLayout(directory), readInstructions);
+    }
+
+    /**
+     * Creates a partitioned table via the flat parquet files from the root {@code directory} using the provided
+     * {@code tableDefinition}.
+     *
+     * @param directory the source of {@link ParquetTableLocationKey location keys} to include
+     * @param readInstructions the instructions for customizations while reading
+     * @param tableDefinition the table definition
+     * @return the table
+     * @see #readPartitionedTable(TableLocationKeyFinder, ParquetInstructions, TableDefinition)
+     * @see ParquetFlatPartitionedLayout#ParquetFlatPartitionedLayout(File)
+     */
+    public static Table readFlatPartitionedTable(
+            @NotNull final File directory,
+            @NotNull final ParquetInstructions readInstructions,
+            @NotNull final TableDefinition tableDefinition) {
+        return readPartitionedTable(new ParquetFlatPartitionedLayout(directory), readInstructions, tableDefinition);
+    }
+
+    /**
+     * Creates a single table via the parquet {@code file} using the table definition derived from that {@code file}.
+     *
+     * <p>
+     * Callers wishing to be more explicit (for example, to skip some columns) may prefer to call
+     * {@link #readSingleTable(File, ParquetInstructions, TableDefinition)}.
+     *
+     * @param file the parquet file
+     * @param readInstructions the instructions for customizations while reading
+     * @return the table
+     * @see ParquetTableLocationKey#ParquetTableLocationKey(File, int, Map)
+     * @see #readSingleFileTable(ParquetTableLocationKey, ParquetInstructions, TableDefinition)
+     */
+    public static Table readSingleTable(
```

Comment:
`readSingleFileTable`?

Reply:
I do think "File" is a bit redundant here; but I could also argue that same point for "Table". I was following the existing nomenclature wrt `readTable`; but it might make sense to go to:

`readSingle`, `readFlatPartitioned`, `readKeyValuePartitioned`.

Reply:
SingleTable seems weird. All of these return a "single table". 
"Single location table" might be more accurate, if you don't want to say "file".

Reply:
I've acquiesced (after looking up how to spell...).